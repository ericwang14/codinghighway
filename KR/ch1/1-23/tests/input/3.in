
#include <stdio.h>

/**/
/***/
/*#include<xpto.h>#/#*|*/
 
size_t size_t_unit() {
	/* sizeof(char) is always 1 */
	size_t res = 0;
	unsigned char c = ~0;
	for (; c; res++, c >>= 1);
	return res;
}

void printbinary(int n) {
	int i/*, a, b, ...*/;
	size_t bits = sizeof(int)*size_t_unit();
	unsigned int j = 1 << (bits-1);
	for (i = 0; i < bits; i++, j >>= 1) {
		printf("%d", (j&n ? 1 : 0));
	}
	printf("\\\n");
}

int main() {
	int c, d;
	c = d = 0;
	printbinary(~0);
	printf("\\/**/ and there was light\\\\"/* this is a comment*/);
	if (c ==/*checking for d/c where c is 0*/ 0)
		printf("Should I say \"Houston, we got a problem?\"");
	else /* everything ok */
		d = d/ /*c*/c;
	printf("/* Life runs on code */\n");
	return 0;
}
