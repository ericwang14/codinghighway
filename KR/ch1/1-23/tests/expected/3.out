
#include <stdio.h>




 
size_t size_t_unit() {
	
	size_t res = 0;
	unsigned char c = ~0;
	for (; c; res++, c >>= 1);
	return res;
}

void printbinary(int n) {
	int i;
	size_t bits = sizeof(int)*size_t_unit();
	unsigned int j = 1 << (bits-1);
	for (i = 0; i < bits; i++, j >>= 1) {
		printf("%d", (j&n ? 1 : 0));
	}
	printf("\\\n");
}

int main() {
	int c, d;
	c = d = 0;
	printbinary(~0);
	printf("\\/**/ and there was light\\\\");
	if (c == 0)
		printf("Should I say \"Houston, we got a problem?\"");
	else 
		d = d/ c;
	printf("/* Life runs on code */\n");
	return 0;
}
